spec/output/generator_spec_rust/union.x/MyXDR.rs

```diff
@@ -0,0 +1,779 @@
+                let i: i32 = (*self).into();
+                i.write_xdr(w)
+            }
+        }
+
+// MyUnion is an XDR Union defines as:
+//
+//   union MyUnion switch (UnionKey type)
+//    {
+//        case ERROR:
+//            Error error;
+//        case MULTI:
+//            Multi things<>;
+//    
+//    
+//    };
+//
+// union with discriminant UnionKey
+#[derive(Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub enum MyUnion {
+  Error(i32),
+  Multi(VecM::<i32>),
+}
+
+        impl MyUnion {
+            #[must_use]
+            pub fn discriminant(&self) -> UnionKey {
+                #[allow(clippy::match_same_arms)]
+                match self {
+                    Self::Error(_) => UnionKey::Error,
+Self::Multi(_) => UnionKey::Multi,
+                }
+            }
+        }
+
+        impl ReadXdr for MyUnion {
+            #[cfg(feature = "std")]
+            fn read_xdr(r: &mut impl Read) -> Result<Self> {
+                let dv: UnionKey = <UnionKey as ReadXdr>::read_xdr(r)?;
+                #[allow(clippy::match_same_arms, clippy::match_wildcard_for_single_variants)]
+                let v = match dv {
+                    UnionKey::Error => Self::Error(i32::read_xdr(r)?),
+UnionKey::Multi => Self::Multi(VecM::<i32>::read_xdr(r)?),
+                    #[allow(unreachable_patterns)]
+                    _ => return Err(Error::Invalid),
+                };
+                Ok(v)
+            }
+        }
+
+        impl WriteXdr for MyUnion {
+            #[cfg(feature = "std")]
+            fn write_xdr(&self, w: &mut impl Write) -> Result<()> {
+                self.discriminant().write_xdr(w)?;
+                #[allow(clippy::match_same_arms)]
+                match self {
+                    Self::Error(v) => v.write_xdr(w)?,
+Self::Multi(v) => v.write_xdr(w)?,
+                };
+                Ok(())
+            }
+        }
+
+// IntUnion is an XDR Union defines as:
+//
+//   union IntUnion switch (int type)
+//    {
+//        case 0:
+//            Error error;
+//        case 1:
+//            Multi things<>;
+//    
+//    };
+//
+// union with discriminant i32
+#[derive(Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub enum IntUnion {
+  V0(i32),
+  V1(VecM::<i32>),
+}
+
+        impl IntUnion {
+            #[must_use]
+            pub fn discriminant(&self) -> i32 {
+                #[allow(clippy::match_same_arms)]
+                match self {
+                    Self::V0(_) => 0,
+Self::V1(_) => 1,
+                }
+            }
+        }
+
+        impl ReadXdr for IntUnion {
+            #[cfg(feature = "std")]
+            fn read_xdr(r: &mut impl Read) -> Result<Self> {
+                let dv: i32 = <i32 as ReadXdr>::read_xdr(r)?;
+                #[allow(clippy::match_same_arms, clippy::match_wildcard_for_single_variants)]
+                let v = match dv {
+                    0 => Self::V0(i32::read_xdr(r)?),
+1 => Self::V1(VecM::<i32>::read_xdr(r)?),
```

Comment:
Minor issue but these seem mis-indented

Reply:
It might be possible for me to fix this a little, but we don't generally try to make xdrgen produce formatted code, and instead we run its output through the languages code formatter outside of xdrgen.